### PR TITLE
Add native LoRA and NF4 QLoRA layers

### PR DESCRIPTION
**Pull request by surya-mp**
## Summary

Adds native parameter-efficient fine-tuning support to GoMLX through a new `ml/layers/peft` package.

The implementation provides:

  - Native LoRA layers for existing GoMLX dense projections.
  - NF4 QLoRA layers with optional double quantization of block scales.
  - Target-module injection through small architecture-facing interfaces.
  - Frozen base projections with adapter-only trainable variable selection.
  - Multiple named adapters sharing one base projection.
  - Runtime adapter selection, composition, and disabling.
  - Configurable LoRA rank, alpha, dropout, target modules, bias behavior, and dense-weight layouts.
  - Validation and rollback handling when host module replacement fails.
  - Detailed usage documentation in `docs/peft.md`.
  - README entries under News and Highlights.

  ## Design

The package keeps GoMLX-specific PEFT behavior close to the graph and variable APIs:

  - `Linear` provides LoRA over standard GoMLX dense weights.
  - `NF4Linear` provides QLoRA over packed NF4 base weights.
  - `Inject` and `InjectNF4` let model architectures expose named modules and replace only configured
  targets.
  - Each returned adapter exposes its own optimizer variable set, so training can update only adapter A/B
  matrices and configured biases.

## Validation

  ```bash
  go test ./ml/layers/peft
```

The focused test suite covers LoRA and QLoRA forward paths, frozen base weights, adapter-only gradients, named adapter selection, target injection, validation, and rollback behavior.

### CODE REVIEW

**Comment by janpfeifer** [comment]
Hi @surya-mp  

Thank you for the contribution! I've been meaning to do this for a while and never had the time. This is very welcome! 

Unfortunately, I'm in a couple of weeks trip without a laptop ... But I'll review/merge it as soon as I'm back, 1st week of October. Apologies.

Cheers

PS: for a later time: this is a useful use case, it could be interesting to create a page in gomlx.github.io, if it's of interest to you. 

I'd also love to add/improve fused ops to optimize the execution of Lora models with the Go backend.
